File: Exemplos/App_eXperience_Java/Elgin_AppExperience_Java-font/app/src/main/java/com/elgin/elginexperience/Services/PrinterService.java
package com.elgin.elginexperience.Services;

import android.app.Activity;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Matrix;
import android.util.Base64;

import com.elgin.e1.Impressora.Termica;

import java.io.File;
import java.util.Map;
import java.util.Objects;

public class PrinterService {
    public Activity mActivity;

    //Variavel utilizada para a verificação de se a conexão atual é por impressora interna ou não
    public boolean isPrinterInternSelected;

    public PrinterService(Activity activity) {
        this.mActivity = activity;
        Termica.setContext(mActivity);
    }

    // Construtor sem argumento utilizado apenas para a criação de testes de validaçõ dos parâmetros.
    public PrinterService() {

    }

    public int printerInternalImpStart() {
        printerStop();
        int result = Termica.AbreConexaoImpressora(6, "M8", "", 0);

        if (result == 0) this.isPrinterInternSelected = true;

        return result;
    }

    public int printerExternalImpStartByIP(String model, String ip, int port) {
        Objects.requireNonNull(model, "model");
        Objects.requireNonNull(ip, "ip");

        printerStop();
        try {
            int result = Termica.AbreConexaoImpressora(3, model, ip, port);

            if (result == 0) this.isPrinterInternSelected = false;

            return result;
        } catch (Exception e) {
            return printerInternalImpStart();
        }
    }

    public int printerExternalImpStartByUSB(String model) {
        Objects.requireNonNull(model);

        printerStop();
        try {
            int result = Termica.AbreConexaoImpressora(1, model, "USB", 0);

            if (result == 0) this.isPrinterInternSelected = false;

            return result;
        } catch (Exception e) {
            return printerInternalImpStart();
        }
    }

    public void printerStop() {
        Termica.FechaConexaoImpressora();
    }

    public int AvancaLinhas(Map<String, Object> map) {
        final int lines = (Integer) Objects.requireNonNull(map.get("quant"), "quant");

        return Termica.AvancaPapel(lines);
    }

    public int cutPaper(Map<String, Object> map) {
        final int lines = (Integer) Objects.requireNonNull(map.get("quant"), "quant");

        return Termica.Corte(lines);
    }

    public int imprimeTexto(Map<String, Object> map) {
        final String text = (String) Objects.requireNonNull(map.get("text"), "text");
        final String align = (String) Objects.requireNonNull(map.get("align"), "align");
        final String font = (String) Objects.requireNonNull(map.get("font"), "font");
        final int fontSize = (Integer) Objects.requireNonNull(map.get("fontSize"), "fontSize");

        int result = 0;
        int alignValue = 0;
        int styleValue = 0;

        // ALINHAMENTO VALUE
        switch (align) {
            case "Esquerda":
                alignValue = 0;
                break;
            case "Centralizado":
                alignValue = 1;
                break;
            default:
                alignValue = 2;
                break;
        }

        //STILO VALUE
        if (font.equals("FONT B")) {
            styleValue += 1;
        }
        if ((boolean) Objects.requireNonNull(map.get("isUnderline"))) {
            styleValue += 2;
        }
        if ((boolean) Objects.requireNonNull(map.get("isBold"))) {
            styleValue += 8;
        }

        result = Termica.ImpressaoTexto(text, alignValue, styleValue, fontSize);
        return result;
    }

    private int codeOfBarCode(String barCodeName) {
        switch (barCodeName) {
            case "UPC-A":
                return 0;
            case "UPC-E":
                return 1;
            case "EAN 13":
            case "JAN 13":
                return 2;
            case "EAN 8":
            case "JAN 8":
                return 3;
            case "CODE 39":
                return 4;
            case "ITF":
                return 5;
            case "CODE BAR":
                return 6;
            case "CODE 93":
                return 7;
            case "CODE 128":
                return 8;
            default:
                throw new AssertionError(barCodeName);
        }
    }

    public int imprimeBarCode(Map<String, Object> map) {
        final int barCodeType = codeOfBarCode((String) Objects.requireNonNull(map.get("barCodeType"), "barCodeType"));
        final String text = (String) Objects.requireNonNull(map.get("text"), "text");
        final int height = (Integer) Objects.requireNonNull(map.get("height"), "height");
        final int width = (Integer) Objects.requireNonNull(map.get("width"), "width");
        final String align = (String) Objects.requireNonNull(map.get("align"), "align");

        final int hri = 4; // NO PRINT
        final int result;
        final int alignValue;

        // ALINHAMENTO VALUE
        switch (align) {
            case "Esquerda":
                alignValue = 0;
                break;
            case "Centralizado":
                alignValue = 1;
                break;
            default:
                alignValue = 2;
                break;
        }

        Termica.DefinePosicao(alignValue);

        result = Termica.ImpressaoCodigoBarras(barCodeType, text, height, width, hri);

        return result;
    }

    public int imprimeQR_CODE(Map<String, Object> map) {
        final int size = (Integer) Objects.requireNonNull(map.get("qrSize"), "qrSize");
        final String text = (String) Objects.requireNonNull(map.get("text"), "text");
        final String align = (String) Objects.requireNonNull(map.get("align"), "align");

        final int nivelCorrecao = 2;
        final int result;
        final int alignValue;

        // ALINHAMENTO VALUE
        switch (align) {
            case "Esquerda":
                alignValue = 0;
                break;
            case "Centralizado":
                alignValue = 1;
                break;
            default:
                alignValue = 2;
                break;
        }

        Termica.DefinePosicao(alignValue);

        result = Termica.ImpressaoQRCode(text, size, nivelCorrecao);
        return result;
    }

    public int imprimeImagem(Bitmap bitmap) {
        //Verifica se o método de impressão atual é por impressora interna ou externa e utiliza a função adequada para cada um
        return isPrinterInternSelected ? Termica.ImprimeBitmap(bitmap) : Termica.ImprimeImagem(bitmap);
    }

    public int imprimeXMLNFCe(Map<String, Object> map) {
        final String xmlNFCe = (String) Objects.requireNonNull(map.get("xmlNFCe"), "xmlNFCe");
        final int indexcsc = (int) Objects.requireNonNull(map.get("indexcsc"), "indexcsc");
        final String csc = (String) Objects.requireNonNull(map.get("csc"), "csc");
        final int param = (int) Objects.requireNonNull(map.get("param"), "param");

        return Termica.ImprimeXMLNFCe(xmlNFCe, indexcsc, csc, param);
    }

    public int imprimeXMLSAT(Map<String, Object> map) {
        final String xml = (String) Objects.requireNonNull(map.get("xmlSAT"), "xmlSAT");
        final int param = (int) Objects.requireNonNull(map.get("param"), "param");

        return Termica.ImprimeXMLSAT(xml, param);
    }

    public int imprimeCupomTEF(Map<String, Object> map) {
        final String base64 = (String) Objects.requireNonNull(map.get("base64"));

        return Termica.ImprimeCupomTEF(base64);
    }

    public int statusGaveta() {return Termica.StatusImpressora(1);}

    public int abrirGaveta() {return Termica.AbreGavetaElgin();}

    public int statusSensorPapel() {
        return Termica.StatusImpressora(3);
    }
}

File: Exemplos/App_eXperience_Kotlin/Elgin_AppExperience_Kotlin-font/app/src/main/java/com/example/e1_kotlin_r/Services/PrinterService.kt
package com.example.e1_kotlin_r.Services

import android.app.Activity
import android.graphics.Bitmap
import com.elgin.e1.Impressora.Termica


class PrinterService(var mActivity: Activity) {
    //Variavel utilizada para a verificação de se a conexão atual é por impressora interna ou não
    var isPrinterInternSelected = false
    fun printerInternalImpStart(): Int {
        printerStop()
        val result = Termica.AbreConexaoImpressora(6, "M8", "", 0)
        if (result == 0) isPrinterInternSelected = true
        return result
    }

    fun printerExternalImpStartByIP(model: String?, ip: String?, port: Int): Int {
        printerStop()
        return try {
            val result = Termica.AbreConexaoImpressora(3, model, ip, port)
            if (result == 0) isPrinterInternSelected = false
            result
        } catch (e: Exception) {
            printerInternalImpStart()
        }
    }

    fun printerExternalImpStartByUSB(model: String?): Int {
        printerStop()
        return try {
            val result = Termica.AbreConexaoImpressora(1, model, "USB", 0)
            if (result == 0) isPrinterInternSelected = false
            result
        } catch (e: Exception) {
            printerInternalImpStart()
        }
    }

    fun printerStop() {
        Termica.FechaConexaoImpressora()
    }

    fun AvancaLinhas(map: Map<*, *>): Int {
        val lines = map["quant"] as Int
        return Termica.AvancaPapel(lines)
    }

    fun cutPaper(map: Map<*, *>): Int {
        val lines = map["quant"] as Int
        return Termica.Corte(lines)
    }

    fun imprimeTexto(map: Map<*, *>): Int {
        val text = map["text"] as String?
        val align = map["align"] as String?
        val font = map["font"] as String?
        val fontSize = map["fontSize"] as Int

        //System.out.println(text + " " + align + " " + font + " " + String.valueOf(fontSize));
        var result = 0
        var alignValue = 0
        var styleValue = 0

        // ALINHAMENTO VALUE
        alignValue = when (align) {
            "Esquerda" -> {
                0
            }
            "Centralizado" -> {
                1
            }
            else -> {
                2
            }
        }
        //STILO VALUE
        if (font == "FONT B") {
            styleValue += 1
        }
        if (map["isUnderline"] as Boolean) {
            styleValue += 2
        }
        if (map["isBold"] as Boolean) {
            styleValue += 8
        }
        result = Termica.ImpressaoTexto(text, alignValue, styleValue, fontSize)
        return result
    }

    private fun codeOfBarCode(barCodeName: String?): Int {
        return if (barCodeName == "UPC-A") 0 else if (barCodeName == "UPC-E") 1 else if (barCodeName == "EAN 13" || barCodeName == "JAN 13") 2 else if (barCodeName == "EAN 8" || barCodeName == "JAN 8") 3 else if (barCodeName == "CODE 39") 4 else if (barCodeName == "ITF") 5 else if (barCodeName == "CODE BAR") 6 else if (barCodeName == "CODE 93") 7 else if (barCodeName == "CODE 128") 8 else 0
    }

    fun imprimeBarCode(map: Map<*, *>): Int {
        val barCodeType = codeOfBarCode(map["barCodeType"] as String?)
        val text = map["text"] as String?
        val height = map["height"] as Int
        val width = map["width"] as Int
        val align = map["align"] as String?
        val hri = 4 // NO PRINT

        // ALINHAMENTO VALUE
        val alignValue: Int = when (align) {
            "Esquerda" -> {
                0
            }
            "Centralizado" -> {
                1
            }
            else -> {
                2
            }
        }
        Termica.DefinePosicao(alignValue)
        return Termica.ImpressaoCodigoBarras(barCodeType, text, height, width, hri)
    }

    fun imprimeQR_CODE(map: Map<*, *>): Int {
        val size = map["qrSize"] as Int
        val text = map["text"] as String?
        val align = map["align"] as String?
        val nivelCorrecao = 2

        // ALINHAMENTO VALUE
        val alignValue: Int = when (align) {
            "Esquerda" -> {
                0
            }

            "Centralizado" -> {
                1
            }

            else -> {
                2
            }
        }
        Termica.DefinePosicao(alignValue)
        return Termica.ImpressaoQRCode(text, size, nivelCorrecao)
    }

    fun imprimeImagem(bitmap: Bitmap?): Int {

        //Verifica se o método de impressão atual é por impressora interna ou externa e utiliza a função adequada para cada um
        val result: Int = if (isPrinterInternSelected) Termica.ImprimeBitmap(bitmap) else {
                Termica.ImprimeImagem(bitmap)
            }
        return result
    }

    fun imprimeXMLNFCe(map: Map<*, *>): Int {
        val xmlNFCe = map["xmlNFCe"] as String?
        println(xmlNFCe)
        val indexcsc = map["indexcsc"] as Int
        val csc = map["csc"] as String?
        val param = map["param"] as Int
        return Termica.ImprimeXMLNFCe(xmlNFCe, indexcsc, csc, param)
    }

    fun imprimeXMLSAT(map: Map<*, *>): Int {
        val xml = map["xmlSAT"] as String?
        val param = map["param"] as Int
        return Termica.ImprimeXMLSAT(xml, param)
    }

    fun imprimeCupomTEF(map: Map<*, *>): Int {
        val base64 = map["base64"] as String?
        return Termica.ImprimeCupomTEF(base64)
    }

    fun statusGaveta(): Int {
        return Termica.StatusImpressora(1)
    }

    fun abrirGaveta(): Int {
        return Termica.AbreGavetaElgin()
    }

    fun statusSensorPapel(): Int {
        return Termica.StatusImpressora(3)
    }

    init {
        Termica.setContext(mActivity)
    }
}
